bobtimus/src/bin/fake_bobtimus.rs

```diff
@@ -39,9 +47,18 @@ async fn main() -> Result<()> {
         .allow_methods(vec!["GET"])
         .allow_header("content-type");
 
-    let latest_rate = warp::path("rate")
-        .and(warp::get())
-        .map(|| warp::sse::reply(warp::sse::keep_alive().stream(fixed_rate_stream())));
+    let latest_rate = warp::path("rate").and(warp::get()).map(|| {
+        warp::sse::reply(
+            warp::sse::keep_alive().stream(rate_service.subscribe().map(|data| {
+                Result::<_, Infallible>::Ok((
+                    warp::sse::event("rate"),
+                    warp::sse::data(
```

Comment:
According to the docs, there is also `warp::sse::json`!

Reply:
https://docs.rs/warp/0.2.5/warp/filters/sse/fn.json.html

Reply:
Perfect, just what I needed.